internal/runtime/runtime.go

```diff
@@ -145,50 +142,54 @@ func (gr *GolangRuntime) ProcessMessage(edgexcontext *appcontext.Context, envelo
 		}
 	}
 
-	edgexcontext.CorrelationID = envelope.CorrelationID
+	appContext.SetCorrelationID(envelope.CorrelationID)
 
 	// All functions expect an object, not a pointer to an object, so must use reflection to
 	// dereference to pointer to the object
 	target = reflect.ValueOf(target).Elem().Interface()
 
 	// Make copy of transform functions to avoid disruption of pipeline when updating the pipeline from registry
 	gr.isBusyCopying.Lock()
-	transforms := make([]appcontext.AppFunction, len(gr.transforms))
+	transforms := make([]interfaces.AppFunction, len(gr.transforms))
 	copy(transforms, gr.transforms)
 	gr.isBusyCopying.Unlock()
 
-	return gr.ExecutePipeline(target, envelope.ContentType, edgexcontext, transforms, 0, false)
+	return gr.ExecutePipeline(target, envelope.ContentType, appContext, transforms, 0, false)
 }
 
-func (gr *GolangRuntime) ExecutePipeline(target interface{}, contentType string, edgexcontext *appcontext.Context,
-	transforms []appcontext.AppFunction, startPosition int, isRetry bool) *MessageError {
+func (gr *GolangRuntime) ExecutePipeline(
+	target interface{},
+	contentType string,
+	appContext *appfunction.Context,
+	transforms []interfaces.AppFunction,
+	startPosition int,
+	isRetry bool) *MessageError {
 
 	var result interface{}
 	var continuePipeline = true
 
-	edgexcontext.SecretProvider = gr.secretProvider
-
 	for functionIndex, trxFunc := range transforms {
 		if functionIndex < startPosition {
 			continue
 		}
 
-		edgexcontext.RetryData = nil
+		appContext.SetRetryData(nil)
 
 		if result == nil {
-			continuePipeline, result = trxFunc(edgexcontext, target, contentType)
+			appContext.SetInputContentType(contentType)
+			continuePipeline, result = trxFunc(appContext, target)
 		} else {
-			continuePipeline, result = trxFunc(edgexcontext, result)
+			continuePipeline, result = trxFunc(appContext, result)
 		}
```

Comment:
it seems to me the initial value for the variable `continuePipeline` doesn't matter at all as its value get override in the if{}else{} block anyway.

Reply:
I think it needs to be declared outside the if/else blocks to be used later on - this would just allow assigning the initial value once instead of once per loop execution if I'm reading right.

Reply:
Correct the assignment of `true` on line 169 isn't needed. Removed assignment. It does need to be assign each iteration of the loop since the next function is call each iteration.